views/js/previewer/plugins/tools/scale/scale.js

```diff
@@ -0,0 +1,532 @@
+                return;
+            }
+
+            re = new RegExp(orientation, 'g');
+            previewFrame = controls.$scaleWrapper.find('.preview-outer-frame')[0];
+            previewFrame.className = previewFrame.className.replace(re, newOrientation);
+            orientation = newOrientation;
+        },
+
+        /**
+         * Configures screen size configuration state according to passed value from affected control
+         * and calls this.scaleFrame() after all.
+         * @param {Event} event - event object related to affected control.
+         * @private
+         */
+        onDeviceChange: function onDeviceChange(event){
+            var element = event.target;
+            var type = selectedDevice;
+            var val = element.value.split(',');
+            var i = val.length;
+            var container = controls.$scaleWrapper.find('.' + type + '-preview-container');
+
+            while (i--) {
+                val[i] = parseFloat(val[i]);
+            }
+            if (type === 'mobile' && orientation === 'portrait') {
+                sizeSettings = {
+                    width: val[1],
+                    height: val[0]
+                };
+            }
+            else {
+                sizeSettings = {
+                    width: val[0],
+                    height: val[1]
+                };
+            }
+
+            if (sizeSettings.width === container.width() && sizeSettings.height === container.height()) {
+                return false;
+            }
+
+            container.css(sizeSettings);
+            this.scaleFrame();
+        },
+
+        /**
+         * Configures selected device's orientation configuration state according to passed value from affected control
+         * and calls this.setOrientation(newOrientation) and this.scaleFrame() after all.
+         * @param {Event} event - event object related to affected control.
+         * @private
+         */
+        onOrientationChange: function onOrientationChange(event) {
+            var element = event.target;
+            var container = controls.$scaleWrapper.find('.' + selectedDevice + '-preview-container');
+            var newOrientation = element.value;
+
+            if (newOrientation === orientation) {
+                return false;
+            }
+            sizeSettings = {
+                height: container.width(),
+                width: container.height()
+            };
+
+            container.css(sizeSettings);
+
+            this.setOrientation(newOrientation);
+            this.scaleFrame();
+        },
+        
+        /**
+         * Configures selected device type on which depends all other parts of the plugin
+         * decides whether it needs to draw device frame around item content and scale it accordingly
+         * @param {Event} event - event object related to affected control.
+         * @private
+         */
+        onDeviceTypeChange: function onDeviceTypeChange(event) {
```

Comment:
Is the event really needed here? Only having the target element would be enough.